provider/role_resource_test.go

```diff
@@ -0,0 +1,210 @@
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testFireHydrantIsSetup(t) },
+		ProviderFactories: defaultProviderFactories(),
+		CheckDestroy:      testAccCheckRoleResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccRoleConfig_basic(rName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttrSet("firehydrant_role.test_role", "id"),
+					resource.TestCheckResourceAttr("firehydrant_role.test_role", "name", fmt.Sprintf("test-role-%s", rName)),
+					resource.TestCheckResourceAttr("firehydrant_role.test_role", "description", "Test role for Terraform"),
+					resource.TestCheckResourceAttrSet("firehydrant_role.test_role", "slug"),
+
+					resource.TestCheckResourceAttr("firehydrant_role.test_role", "built_in", "false"),
+					resource.TestCheckResourceAttr("firehydrant_role.test_role", "permissions.#", "1"),
+				),
+			},
+			// Test update
+			{
+				Config: testAccRoleConfig_updated(rName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttrSet("firehydrant_role.test_role", "id"),
+					resource.TestCheckResourceAttr("firehydrant_role.test_role", "name", fmt.Sprintf("updated-role-%s", rName)),
+					resource.TestCheckResourceAttr("firehydrant_role.test_role", "description", "Updated test role"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccRoleResource_withPermissions(t *testing.T) {
+	rName := acctest.RandStringFromCharSet(20, acctest.CharSetAlphaNum)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testFireHydrantIsSetup(t) },
+		ProviderFactories: defaultProviderFactories(),
+		CheckDestroy:      testAccCheckRoleResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccRoleConfig_withPermissions(rName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttrSet("firehydrant_role.test_role", "id"),
+					resource.TestCheckResourceAttr("firehydrant_role.test_role", "name", fmt.Sprintf("test-role-%s", rName)),
+					resource.TestCheckResourceAttr("firehydrant_role.test_role", "permissions.#", "20"),
+				),
+			},
+			// Update permissions
+			{
+				Config: testAccRoleConfig_withUpdatedPermissions(rName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttrSet("firehydrant_role.test_role", "id"),
+					resource.TestCheckResourceAttr("firehydrant_role.test_role", "permissions.#", "19"),
+				),
+			},
+		},
+	})
+}
+
+func testAccRoleConfig_basic(rName string) string {
+	return fmt.Sprintf(`
+	resource "firehydrant_role" "test_role" {
+		name        = "test-role-%s"
+		description = "Test role for Terraform"
+		permissions = [
+			"read_users"
+		]
+	}
+	`, rName)
+}
+
+func testAccRoleConfig_updated(rName string) string {
+	return fmt.Sprintf(`
+	resource "firehydrant_role" "test_role" {
+		name        = "updated-role-%s"
+		description = "Updated test role"
+		permissions = [
```

Comment:
We're updating name and description; can we update the permissions here also?  That's probably going to be the most common use case.